src/lib.rs

```diff
@@ -306,13 +306,25 @@ impl KVMIntrospectable for KVMi {
         Ok(vcpu_count)
     }
 
-    fn get_registers(&self, vcpu: u16) -> Result<(kvm_regs, kvm_sregs, kvm_msrs), Error> {
+    fn get_registers(&self, vcpu: u16) -> Result<(kvm_regs, kvm_sregs, KvmMsr), Error> {
         let mut regs: kvm_regs = unsafe { mem::MaybeUninit::<kvm_regs>::zeroed().assume_init() };
         let mut sregs: kvm_sregs = unsafe { mem::MaybeUninit::<kvm_sregs>::zeroed().assume_init() };
-        let mut msrs: kvm_msrs = unsafe { mem::MaybeUninit::<kvm_msrs>::zeroed().assume_init() };
+        let mut msrs: KvmMsr = unsafe { mem::MaybeUninit::<KvmMsr>::zeroed().assume_init() };
         let mut mode: c_uint = 0;
+        msrs.msrs.nmsrs = 6;
+        msrs.entries[0].index = KVMiMsr::SysenterCs as u32;
```

Comment:
Your are working around the problem that you have an `__IncompleteArrayField` in `kvm_msrs`. However putting an allocated array behind a zero length `__IncompleteArrayField` is super confusing imho. The correct way to handle incomplete arrays is by allocating the memory with [std::alloc::from_size_align](https://doc.rust-lang.org/std/alloc/struct.Layout.html). Here is a very similar example: https://github.com/rust-lang/rust-bindgen/issues/1680
If you implement it this way you don't need your custom `KvmMsr` anymore.